### ISSUE TITLE
Using multi build on local source dir is not working

### ISSUE THREAD

**Issue by AnimMouse** [rdp/ffmpeg-windows-build-helpers#469]
Using multi with --ffmpeg-source-dir
```
./cross_compile_ffmpeg.sh --ffmpeg-source-dir=$GITHUB_WORKSPACE/ffmpeg-source --gcc-cpu-count=2 --disable-nonfree=n --sandbox-ok=y --compiler-flavors=multi
```
Returns with
```
2020-07-14T20:43:15.0520578Z WARNING: We thought we found pkg-config '/home/runner/work/ffmpeg-autobuild/ffmpeg-autobuild/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-pkg-config' but now it's not there. How odd!
2020-07-14T20:43:15.0520834Z Found Pkg-config: NO
2020-07-14T20:43:15.0520941Z Found CMake: NO
2020-07-14T20:43:15.0521181Z Run-time dependency libffi found: NO (tried pkgconfig and cmake)
2020-07-14T20:43:15.0521297Z Library ffi found: YES
2020-07-14T20:43:15.0521397Z Has header "ffi.h" : YES 
2020-07-14T20:43:15.0521633Z Run-time dependency zlib found: YES 1.2.11
2020-07-14T20:43:15.0521744Z Checking for function "ngettext" : NO 
2020-07-14T20:43:15.0521848Z Library intl found: YES
2020-07-14T20:43:15.0522110Z Checking for function "bind_textdomain_codeset" with dependency -lintl: NO 
2020-07-14T20:43:15.0522233Z Library ws2_32 found: YES
2020-07-14T20:43:15.0522326Z Checking for function "strlcpy" : NO 
2020-07-14T20:43:15.0522434Z Program python3 found: YES (/usr/bin/python3)
2020-07-14T20:43:15.0522543Z Program bash found: YES
2020-07-14T20:43:15.0522643Z Program sh found: YES
2020-07-14T20:43:15.0522742Z Program env found: YES
2020-07-14T20:43:15.0522912Z Configuring glibconfig.h using configuration
2020-07-14T20:43:15.0523022Z Fetching value of define "arm" :  
2020-07-14T20:43:15.0523115Z Fetching value of define "__arm" :  
2020-07-14T20:43:15.0523222Z Fetching value of define "__arm__" :  
2020-07-14T20:43:15.0523506Z Compiler for C supports arguments -Wno-format-nonliteral -Wformat-nonliteral: YES 
2020-07-14T20:43:15.0523797Z Compiler for C supports arguments -Wno-duplicated-branches -Wduplicated-branches: YES 
2020-07-14T20:43:15.0523881Z 
2020-07-14T20:43:15.0523997Z glib/gnulib/meson.build:277:2: ERROR: Problem encountered: frexp() is missing or broken beyond repair, and we have nothing to replace it with
2020-07-14T20:43:15.0524090Z 
2020-07-14T20:43:15.0524396Z A full log can be found at /home/runner/work/ffmpeg-autobuild/win64/glib-2.64.3/build/meson-logs/meson-log.txt
2020-07-14T20:43:15.1377176Z ##[error]Process completed with exit code 1.
```
[Full Log.txt.gz](https://github.com/rdp/ffmpeg-windows-build-helpers/files/4942168/Full.Log.txt.gz)
Running this script on GitHub Actions with Ubuntu 18.04

**Comment by PeacefullyDisturbed** [comment]
I was able to reproduce this issue trying a multi build with a ffmpeg source directory on both Ubuntu 18.04 and 20.04.

It it because the build_ffmpeg function simply does `cd ..` at the end, so after building the win32 ffmpeg the working directory is now one level above wherever the ffmpeg source directory is.  Since this likely isn't in the sandbox when using a separate ffmpeg source directory the win64 stuff would all be build in the wrong place, so eventually some things could fail.  In particular the Meson builds would try using the win32 cross file as if the win64 one was made it would be in the wrong place.

This can also clobber stuff outside of the sandbox which isn't good either.

I've submitted a pull-request to fix this (#478); if you want to test it before it's merged you can simply apply the patch file yourself (https://github.com/rdp/ffmpeg-windows-build-helpers/pull/478.patch).

**Comment by AnimMouse** [comment]
Thanks!
